Opentrons/buildroot
# boards/opentrons/ot2: Fix /etc/machine-id changing each reboot.

# Overview

Fixes #121.

See the in-script comments for a writeup of the problem and solution approach. I've tried to keep them brief-ish and self-contained.

# Manual testing

1. Flash ot2-fullimage.zip.
2. Boot.
3. `cat /etc/machine-id` and note the output.
4. Soft-reboot, power cycle, etc. a bunch of times. Each time, check `cat /etc/machine-id`. It should never change.
5. Check `/var/log/journal`. It should have only two directories: one named after the machine-id, and one named `remote`.
6. Send an update to the OT-2 through the Opentrons App. It can be any relatively recent ot2-system.zip.
7. `cat /etc/machine-id` once more. It should still be the same.

# Alternative solutions

See https://github.com/Opentrons/software-factory-scripts/issues/36 for other possible solutions (and non-solutions) that I decided not to go with.
